waddrmgr/manager.go

```diff
@@ -532,6 +532,61 @@ func (m *Manager) NewScopedKeyManager(ns walletdb.ReadWriteBucket, scope KeyScop
 	return m.scopedManagers[scope], nil
 }
 
+// NewWatchingOnlyScopedKeyManager creates a new watching-only scoped key manager
+// from the root manager.
+// A scoped key manager is a sub-manager that only has the coin type key of a
+// particular coin type and BIP0043 purpose. This is useful as it enables
+// callers to create an arbitrary BIP0043 like schema with a stand alone
+// manager.
+func (m *Manager) NewWatchingOnlyScopedKeyManager(ns walletdb.ReadWriteBucket, scope KeyScope,
```

Comment:
This could be merged with `NewScopedKeyManager`, with a conditional on `watchingOnly`.  WDYT?

Reply:
SGTM, there's a ton of duplication as is.

Reply:
Yes, that would definitely be preferable. You could split the content into multiple smaller methods and call them depending on the flag. This would avoid large chunks of code inside if conditions.

Reply:
Fixed.

Reply:
Let us know if looks good.